packages/forms/src/elements/search/search.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState, useEffect } from 'react';
+import { Field, FieldRenderProps } from 'react-final-form';
+import { Flex } from '@tpr/core';
+import { StyledInputLabel, InputElementHeading } from '../elements';
+import { FieldProps, FieldExtraProps } from '../../renderFields';
+import AccessibilityHelper from '../accessibilityHelper';
+import Autocomplete from 'accessible-autocomplete/react';
+import { filterResults, formatItemDefault } from './filterResults';
+import { act } from 'react-dom/test-utils';
+import styles from './search.module.scss';
+
+interface SearchProps extends FieldRenderProps<string>, FieldExtraProps {
+	callback?: Function;
+	formatItem?: (item: any) => string;
+	getSelectedItem?: (item: any) => string;
+	keyValue: string;
+	notFoundMessage?: string;
+	optionsArray?: any[];
+	searchService?: (x: string) => Promise<any>;
+}
+
+type PanelVisibility = 'visible' | 'hidden' | 'complete';
+
+const Search: React.FC<SearchProps> = React.memo(
+	({
+		callback,
+		cfg,
+		formatItem = formatItemDefault,
+		getSelectedItem,
+		hint,
+		inputWidth = 10,
+		keyValue,
+		label,
+		meta,
+		name,
+		notFoundMessage = 'Your search criteria has no match',
+		optionsArray = [],
+		placeholder,
+		required = true,
+		searchService,
+		testId = 'search',
+		...rest
+	}) => {
+		const [panelVisible, setPanelVisible] = useState<PanelVisibility>('hidden');
+		const [classes, setClasses] = useState(styles.autocomplete);
+		const [optionsArrayObjects, setOptionsArrayObjects] = useState([]);
+
+		const getSelectedItemDefault = (item) => {
+			return item && item[keyValue];
+		};
+
+		const saveResultsInState = (resultsFiltered) => {
+			act(() => setOptionsArrayObjects(resultsFiltered));
+			if (panelVisible == 'hidden') act(() => setPanelVisible('visible'));
+		};
+
+		const showResultsFromOptionsArray = (query: string): any[] => {
+			const results = filterResults(query, optionsArray, keyValue);
+			saveResultsInState(results);
+			return results;
+		};
+
+		const showResultsFromSearchService = (apiResponse): any[] => {
+			saveResultsInState(apiResponse);
+			return apiResponse;
+		};
+
+		const getResults = async (query: string, populateResults: Function) => {
+			searchService
+				? await searchService(query).then((response) =>
+						populateResults(
+							query ? showResultsFromSearchService(response) : [],
+						),
+				  )
+				: populateResults(query ? showResultsFromOptionsArray(query) : []);
+		};
+
+		const toggleResultsPanel = () => {
+			const newClasses =
```

Comment:
I agree with the SonarCloud bot here - while this works, it would be a lot clearer without the nested ternary operator.

Reply:
I have removed the nested ternary operator for clarity.